klippa/src/hmtx.rs

```diff
@@ -60,15 +72,25 @@ pub fn subset_hmtx_hhea(
     Ok(())
 }
 
-fn compute_new_num_h_metrics(hmtx: &Hmtx, gids: &IntSet<GlyphId>) -> usize {
-    let num_long_metrics = gids.len().min(0xFFFF) as usize;
-    let last_gid = gids.last().unwrap();
-    let last_advance = hmtx.advance(last_gid).unwrap();
+fn compute_new_num_h_metrics(
+    hmtx: &Hmtx,
+    gid_set: &IntSet<GlyphId>,
+    num_output_glyphs: usize,
+) -> usize {
+    let mut num_long_metrics = num_output_glyphs.min(0xFFFF) as u32;
+    let last_gid = num_long_metrics - 1;
+    let last_advance = hmtx.advance(GlyphId::from(last_gid)).unwrap();
 
-    let num_skippable_glyphs = gids
-        .iter()
-        .rev()
-        .take_while(|gid| hmtx.advance(*gid).unwrap() == last_advance)
-        .count();
-    (num_long_metrics - num_skippable_glyphs + 1).max(1)
+    while num_long_metrics > 1 {
+        let gid = GlyphId::from(num_long_metrics - 2);
+        let mut advance = 0;
+        if gid_set.contains(gid) {
+            advance = hmtx.advance(gid).unwrap();
+        }
```

Comment:
why mutate advance? Can't we just set it directly, like

```rust
let advance = gid_set.contains(gid)
    .then(|| hmtx.advance(gid).unwrap())
    .unwrap_or(0);
```

Reply:
new learning:) Thanks!